reverseArray.js

```diff
@@ -0,0 +1,20 @@
+function transformArrayOfNumbers(array) {
+    //fliter and return odd numbers
+   const oddNumbers =  array.filter( (arrNumber)=> {
+        return arrNumber %2 == 1
+    })
+
+    //an empty array to store doubled numbers in reverse
+  const doubledNumbers = [] 
+
+  oddNumbers.forEach((no) => {
+    doubledNumbers.unshift( no * 2)
```

Comment:
Good knowledge of `unshift` method, and this works but if you do this on a large dataset this can easily become expensive because what is happening is that you are adding new elements to the start of the array and then you have to move all the other existing elements.

You can read more about how unshift works in this [article](https://www.javascripttutorial.net/javascript-array-unshift/) 

A more efficient way to do this would be to store the double numbers first in an array and then reverse that array using the [reverse](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reverse) method.